### PR TITLE
pb/config

### PR DESCRIPTION
**Pull request by pbrisbin**
Changes:

- Introduce JUnitConfig
- Update README for new usage
- Deprecate old HSpec name-space
- Re-export Config module
- Document Config module

Message from the main commit:

Our modules were mis-named ("HSpec" instead of "Hspec"). This presents
an opportunity to change our interface in a minor version bump by doing
so in a (new) correctly-named module space.

This patch does that, and moves from positional arguments to a
JUnitConfig record. The record supports the same behavior but is more
flexible and has been written in an encapsulated style to start. Names
were chosen semi-arbitrarily to match jest-junit. Documentation will
follow soon.

This configuration object will be useful as we evolve the reporter's
features; for example we will be adding the ability to specify prefixes
for reported paths (e.g. to support monorepositories). Having this
record will make that a backwards-compatible extension.

The HSpec-named modules have remained as pass-through to the new
name-space, and will be deprecated in the next commit.

### CODE REVIEW

File: library/Test/HSpec/JUnit.hs [hunk 1]
```diff
@@ -1,136 +1,34 @@
 module Test.HSpec.JUnit
+  {-# DEPRECATED "Use Test.Hspec.JUnit instead" #-}
```
Comment (mjgpy3) [inline library/Test/HSpec/JUnit.hs:2]: :+1:

File: library/Test/HSpec/JUnit/Schema.hs [hunk 2]
```diff
@@ -1,35 +1,9 @@
 module Test.HSpec.JUnit.Schema
+  {-# DEPRECATED "Use Test.Hspec.JUnit.Schema instead" #-}
```
Comment (mjgpy3) [inline library/Test/HSpec/JUnit/Schema.hs:2]: :+1:

**Review by mjgpy3 — APPROVED** [review]
Looks good! Thanks for adding the docs and `DEPRECATED` pragmas.

I noticed you have a couple of commits prefixed `fixup!` -- were these meant to be squashed?

**Comment by pbrisbin** [comment]
> I noticed you have a couple of commits prefixed fixup! -- were these meant to be squashed?

Yeah, I always push with history as-is for the reviewer, then clean it up after review. Thinking being, you might have comments to address as other fixups, etc. So I do history cleanup post-review / pre-merge. Kind of annoying if/when the history itself could be something worth reviewing, but I haven't found a great answer to that.

**Pull request merged**